examples/hello/tasks/helloCall.ts

```diff
@@ -49,7 +55,7 @@ const main = async (args: any, hre: HardhatRuntimeEnvironment) => {
     ethers.utils.concat([functionSignature, encodedParameters])
   );
 
-  const gasLimit = hre.ethers.BigNumber.from(args.gasLimit);
+  const gasLimit = hre.ethers.BigNumber.from(args.txOptionsGasLimit);
   const zrc20 = new ethers.Contract(args.zrc20, ZRC20ABI.abi, signer);
   const [, gasFee] = await zrc20.withdrawGasFeeWithGasLimit(gasLimit);
   const zrc20TransferTx = await zrc20.transfer(
```

Comment:
@skosito @lumtis I'm still confused about what is the best practice for calling a universal app's function that calls the gateway: does a user need to send ZRC-20s for gas to the contract (like implemented here) or can/should the user approve the universal app to spend the required ZRC-20?

Reply:
Never mind, i figured it out. Both options are possible and the approve method provides better UX.